contracts/swapHandlers/SwapHandlerBase.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./ISwapHandler.sol";
+import "../Interfaces.sol";
+import "../Utils.sol";
+
+/// @notice Base contract for swap handlers
+abstract contract SwapHandlerBase is ISwapHandler {
+    function setMaxAllowance(address token, address spender) internal {
+        uint allowance = IERC20(token).allowance(address(this), spender);
+        if (allowance == 0) Utils.safeApprove(token, spender, type(uint).max);
```

Comment:
wouldn’t it be better if comparison was made against amountIn rather than 0? if allowance were set to max, it'll probably be enough to make the swaps 'forever', but for the correctness sake, it'd compare against amountIn. what do you think?

Reply:
I think @kasperpawlowski 's suggestion makes sense -- it should only need to increase the approval if it's currently insufficient for the max in.

Actually, calling `allowance` is probably roughly the same gas cost as just calling `safeApprove` (at least in the case of "maxable" tokens). So maybe it is easier to just *always* set the approval?

Reply:
I run some tests. In base scenario the allowance is set externally and not touched during the swaps. Testing **second** swap of the asset, on a cleaned up TestERC20.approve (no test behaviours).
base: 260599
A) +1311 - set max allowance if < amountIn (Kasper's suggestion)
B) +3902 - always set max.  I think the cost is mostly in the Approval event emitted every time.
C) +668 - set allowance when requested offchain through an additional SwapPrams.setAllowance

I thought we'd get better savings with C, but imo A strikes the best balance.